### PR TITLE
feat(dependencies): Upgrade  django-phonenumber-field to v7

### PR DESCRIPTION
**Pull request by mlec1**
## Description
Upgrade django-phonenumber-field to v7

Refs:
- https://github.com/jazzband/django-two-factor-auth/issues/639

## Motivation and Context
Refs:
- https://github.com/jazzband/django-two-factor-auth/issues/639

## How Has This Been Tested?
Locally tested with the example app + run the unit tests successfully

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
         'Django>=3.2',
         'django_otp>=0.8.0',
         'qrcode>=4.0.0,<7.99',
-        'django-phonenumber-field>=1.1.0,<7',
+        'django-phonenumber-field>=1.1.0,<8',
```
Comment (claudep) [inline setup.py:18]: Considering that version 1.1.0 was released in 2016, do you think we can drop the `>=1.1.0` part?

Comment (mlec1) [reply setup.py:18]: Well for me, it would be fine. Which version would you like to target `>=2.0.0` or `>=3.0.0` or more

Comment (claudep) [reply setup.py:18]: Simply don't provide a minimal version, unless it's required by some specific change.

**Review by claudep — APPROVED** [review]
Thanks!

**Pull request merged**